### PR TITLE
Feat/501 bop map

### LINKED ISSUE #501: Add Map to BOP (Federal) Page
## Objective
Add a map to the BOP page in order to make designs uniform to state and ICE pages

- This map will correlate to the facilities map on state pages
- Map will essentially be a replication of a filtered down homepage map (see screenshot below) 

## Answered Questions
- Do we have the location data to correlated federal facilities to a map? Yes, we have lat/lon location data, no data to link to counties though. This could potentially be the homepage map (filtered down to Federal facilities) replicated. 
- Will we need a map next to the releases section as is the case with state pages? Since there are only 4 federal releases, this is will not be necessary. 
- Should we update the facilities table with a tooltip to view facility names? Yes, we will update the styling on the BOP facilities page to have a uniform tooltip to other areas in the site where facility name needs to be clarified/fully viewed. 

## Workflow Checklist
- [x] [UCLA] Provide feature requirements
- [x] [HO] Draft feature specification
- [x] [UCLA] Approve feature specification
- [x] [HO] Mockup designs and plan of approach
- [x] [UCLA] Feedback round (feedback on how closely designs match agreed upon feature spec) 
- [x] [HO] Implement design feedback
- [x] [UCLA] Approve designs
- [x] [HO] Dev implementation demo (iterative) 
- [x] [UCLA] Provide feedback (on ways in which dev implementation differs from designs and agreed upon functionality) 
- [ ] [HO] Final dev implementation demo
- [ ] [UCLA] Approve feature implementation for prod release

### PR DESCRIPTION
**Pull request by hoshmn**
fixes #501

### CODE REVIEW

File: src/templates/federal.js [hunk 1]
```diff
@@ -92,23 +112,28 @@ const useStyles = makeStyles((theme) => ({
         },
       },
     },
-    // HACK: filings overrides, need to restyle this page but this is here for now
-    "& .filings": {
-      width: `54rem`,
-      "& .MuiBox-root .MuiBox-root": { marginLeft: `2rem`, transform: "none" },
```
Comment (hoshmn) [inline src/templates/federal.js:98]: updated Filings component styles, as this is that component is only used here now

File: src/templates/federal.js [hunk 2]
```diff
@@ -92,23 +112,28 @@ const useStyles = makeStyles((theme) => ({
         },
       },
     },
-    // HACK: filings overrides, need to restyle this page but this is here for now
-    "& .filings": {
-      width: `54rem`,
```
Comment (hoshmn) [inline src/templates/federal.js:97]: moved to md+ breakpoint, as it was breaking smaller layouts

File: src/cms/lang/sections/map.js [hunk 3]
```diff
@@ -6,7 +6,6 @@ const map = {
   widget: "object",
   fields: [
     makeStringField("Title"),
-    makeTextField("Description"),
```
Comment (hoshmn) [inline src/cms/lang/sections/map.js:9]: not used

File: src/common/utils/selectors.js [hunk 4]
```diff
@@ -1,38 +1,44 @@
-import { JURISDICTIONS, JURISDICTION_COLORS, KEYS, METRICS } from "../constants"
```
Comment (hoshmn) [inline src/common/utils/selectors.js:1]: in this file just getGradientForJurisdiction added (and lots of semicolons)

File: src/components/controls/JurisdictionToggles.js [hunk 5]
```diff
@@ -54,48 +63,56 @@ const JurisdictionToggles = ({ classes, marker = "spike", ...props }) => {
...
       {...props}
     >
-      {categories.map((c, i) => {
+      {displayedCategories.map((c) => {
         return (
           <FormControlLabel
             className={clsx("toggle", "toggle-" + c)}
             key={c}
             value={c}
+            disabled={singleCategory}
```
Comment (hoshmn) [inline src/components/controls/JurisdictionToggles.js:89]: does a disabled form work here semantically, or should a single category be rendered as a <li>?

**Comment by Lane** [comment]
@hoshmn seeing some conflicts here that is blocking the deploy preview, can you fix that up?

**Comment by hoshmn** [comment]
> @hoshmn seeing some conflicts here that is blocking the deploy preview, can you fix that up?

@Lane sorry about that! fixed

**Comment by Lane** [comment]
@hoshmn looking good, a couple things to fix up:

- the map looks like it's in a nested block, so the margins do not align with the other sections.  can you drop the padding on the inner block?



- the tests metrics result in a blank map.  is there tests data at the federal facility level?  if not, lets drop it from the options.

**Comment by hoshmn** [comment]
@lane 
>     * the map looks like it's in a nested block, so the margins do not align with the other sections.  can you drop the padding on the inner block?

will do!  

>     * the tests metrics result in a blank map.  is there tests data at the federal facility level?  if not, lets drop it from the options.

hmmm looking at the data it appears that most of the federal facility rows (99 of 115) do have a `Residents.Tested` value. What we look for, though, is `Residents.Tadmin` (and that's the column that's used for state facilities)... Should we raise this discrepancy to UCLA and they can fix (or, if needed, we can look in that column for federal facilities)?

**Comment by hoshmn** [comment]
@Lane also, currently the map always shows spikes for `residents` values, regardless of the sort state of the facilities table below it (I was initially following the template of the ICE page, which doesn't have any `staff` values, and the home page map also always shows `residents` values). Should I make it so that the `group` plotted in the map is linked to the facilities table below it? Do we also then need to add a `group` selection dropdown to the MetricSelectionTitle?

**Comment by Lane** [comment]
@hoshmn closing this PR and re-opening with `development` as the base so that the deploy preview generates

**Pull request closed without merging**